File: array/209_MinimumSizeSubarraySum/main.py
from typing import List


class Solution:
    def minSubArrayLen(self, s: int, nums: List[int]) -> int:
        cur_sum = 0
        left = 0
        result = 0
        for right in range(0, len(nums)):
            cur_sum += nums[right]
            while cur_sum >= s:
                result = min(result, 1 + right - left) if result else right - left + 1
                cur_sum -= nums[left]
                left += 1

        return result

File: array/209_MinimumSizeSubarraySum/main.cpp
#include "vector"

class Solution {
public:
    int minSubArrayLen(int target, std::vector<int>& nums) {
        int result = 0;
        int left = 0;
        int sum = 0;
        
        for (int right = 0; right < nums.size(); ++right) {
            sum += nums[right];
            if (sum >= target) {
                while (sum >= target && left <= right) {
                    sum -= nums[left++];
                }
                if (result == 0 || result > right - left + 2) {
                    result = right - left + 2;
                }
            }
        }
        return result;
    }
};